apps/mobile/src/components/balance/balance.tsx

```diff
@@ -1,74 +1,53 @@
 import { PrivateText } from '@/components/private-text';
+import { formatCurrency } from '@/utils/currency-formatter';
 
-import { currencyDecimalsMap, currencyNameMap } from '@leather.io/constants';
 import { Money } from '@leather.io/models';
-import { SkeletonLoader, TextProps } from '@leather.io/ui/native';
-import { formatMoneyWithoutSymbol, i18nFormatCurrency } from '@leather.io/utils';
+import { SkeletonLoader, Text, TextProps } from '@leather.io/ui/native';
+import { FormatAmountOptions } from '@leather.io/utils';
 
 const EmptyBalanceDisplay = '-.--';
 
-export function isQuoteCryptoCurrency(balance: Money) {
```

Comment:
I'm glad to see the back of this nasty code, thanks for refactoring and improving it 👍

Reply:
Thanks @pete-watters I've spent a lot of time finding a way for the new formatter to be currency-type-agnostic, so we can use one function for everything. There's still a non-zero chance that at some point we might need to hint it about the currency type, but we'll deal with it when that happens.